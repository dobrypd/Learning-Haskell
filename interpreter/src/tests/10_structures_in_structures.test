struct {
    struct {
        int a;
        int b;
    } left;
    struct {
        int c;
        int d;
    } right;
} root;

int main() {
    root.left.a = 10;
    root.left.b = 20;
    root.right.c = 5;
    root.right.d = 0;
    return root.left.a + root.left.b + root.right.c + root.right.d;
}
