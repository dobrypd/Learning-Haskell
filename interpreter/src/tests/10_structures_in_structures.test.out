Welcome to Kappa interpreter
35
[(fromList [(Ident "root",fromList [(Ident "left",fromList [(Ident "a",10),(Ident "b",20)]),(Ident "right",fromList [(Ident "c",5),(Ident "d",0)])])],fromList [(Ident "main",([],[SStm (ExprS (Sexpr (Eassign [Ident "root",Ident "left",Ident "a"] (Econst (Eint 10))))),SStm (ExprS (Sexpr (Eassign [Ident "root",Ident "left",Ident "b"] (Econst (Eint 20))))),SStm (ExprS (Sexpr (Eassign [Ident "root",Ident "right",Ident "c"] (Econst (Eint 5))))),SStm (ExprS (Sexpr (Eassign [Ident "root",Ident "right",Ident "d"] (Econst (Eint 0))))),SStm (JumpS (SjumpFive (Eplus (Eplus (Eplus (Evar [Ident "root",Ident "left",Ident "a"]) (Evar [Ident "root",Ident "left",Ident "b"])) (Evar [Ident "root",Ident "right",Ident "c"])) (Evar [Ident "root",Ident "right",Ident "d"]))))]))],fromList [])]
